### ISSUE TITLE
CSP (content security policy) breaks some sites when they are launched from an existing web page

### ISSUE THREAD

**Issue by pitsi** [brave/brave-browser#16251]
## Description 
As mentioned in the title, csp (content security policy) breaks some sites when they are launched from an existing page. 
I use a simple page as my new tab and launch the 20+ sites I mostly visit from there. Right now, facebook and cockpit (local page, web console for linux servers) are affected. The error output in brave's console (f12 > console) always looks similar to this

`Refused to execute inline script because it violates the following Content Security Policy directive: "script-src *". Either the 'unsafe-inline' keyword, a hash ('sha256-random numbers and letters'), or a nonce ('nonce-...') is required to enable inline execution.`

This can also be considered a followup of #13929

## Steps to Reproduce
   1. Create a simple web page locally, e.g.
```
<html>
<head>
<title>a simple page</title>
</head>
<body>
<a href="https://www.facebook.com/">Facebook</a>
<a href="https://192.168.1.5:9090">Cockpit</a>
</body>
</html>
```
   2. Open it in brave.
   3. Click on any of the 2 links mentioned.
   4. Notice that the pages appear broken. 
   5. The errors appear in the console (f12 > console)

## Actual result:
Facebook's page appears completely blank. Cockpit's appears like so

![2021-06-05-135106_1278x720_scrot](https://user-images.githubusercontent.com/5880037/120889976-a1b58d00-c608-11eb-96d4-3eba16c578fc.png)

## Expected result:
I assume everyone knows how facebook's main page looks. Cockpit's should appear like so, prompting the user for credentials.

![2021-06-05-135112_1278x720_scrot](https://user-images.githubusercontent.com/5880037/120889982-a712d780-c608-11eb-9cca-5814f674b218.png)

## Reproduces how often: 
Easily.

## Brave version (brave://version info)
Brave	1.25.68 Chromium: 91.0.4472.77 (Official Build) (64-bit)
Revision	1cecd5c8a856bc2a5adda436e7b84d8d21b339b6-refs/branch-heads/4472@{# 1246}
OS	Linux

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Probably
- Can you reproduce this issue with the nightly channel? Probably

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Does the issue resolve itself when disabling Brave Rewards? No
- Is the issue reproducible on the latest version of Chrome? No. Tested on chrome unstable, v93.0.4530.5 as of today.

## Miscellaneous Information:
In order for facebook to appear blank, the user must be already logged in in facebook. If not, it shows the usual facebook login page but the same errors on the console.

Moreover, if the forementioned link leads to any of facebook's subpages, e.g.
```
facebook.com/messages
facebook.com/bravesoftware
```
the page appears blank as well and the same errors appear on the console.

Last but not least, if instead of pressing the link, the user types the url in the address bar, each page opens with no issues and no errors in the console.

**Comment by ryanbr** [comment]
I think this is related too https://github.com/brave/brave-browser/issues/16156

**Comment by pitsi** [comment]
Maybe. If the devs judge that the issues are the same, feel free to close this one.
I forgot to mention that it started happening after the upgrade to 1.25.x, so it must have appeared at the same time as yours. However, I noticed the problem on cockpit yesterday, so I decided to investigate it more and then I opened this issue report here.

**Comment by ryanbr** [comment]
Can you check if Chrome is affected? we've had reports Chrome had similar issues. @pitsi

**Comment by pitsi** [comment]
I have chromium and chrome, both on v91.0.4472.77, but on my windows partition. I will later transfer my page there and check.
On linux, as I have mentioned above, chrome unstable (v93.0.4530.5) is not affected.

**Comment by pitsi** [comment]
And no, chromium and chrome from windows are not affected. Both pages load with no issues and no errors in the console.

**Comment by ask1234560** [comment]
Can confirm, quora is breaking. If the site is opened in the same tab. But it works when the site is opened in new tab.

**Comment by pitsi** [comment]
I do not have a quora account, thus I am not logged in to the site. I added it to my simple page, and it opens up as usual, asking for credentials. If you do have an account in quora, can you please check the console for similar errors?

The forementioned sites (facebok and cockpit) cause the issue whether they are lauched on the same tab or in a new one.

**Comment by pitsi** [comment]
One very minor discovery I just made. Pressing f5 to refresh that broken page (= any of the pages mentioned above) does not resolve the issue. The same errors appear on the console too.
One has to click in the address bar and press enter, as if he typed the full url in the first place, as a workaround.

**Comment by ryanbr** [comment]
Okay looked into this. Bisected to a

* Create a basic .html file
* `<html><body><a href="https://duckduckgo.com/">Link</a></body></html>`
* Open link into Brave
* Fails from `Nightly v1.25.17` onwards.

```
Nightly v1.25.46   Bad
Nightly v1.25.34   Bad
Nightly v1.25.17   Bad
Nightly v1.25.16   Good
Nightly v1.25.15   Good
```

https://github.com/brave/brave-browser/compare/v1.25.16...v1.25.17

**Comment by antonok-edm** [comment]
@pitsi @ask1234560 I'm investigating a fix for this; in the meantime you should be able to work around it by disabling `#brave-adblock-csp-rules` under `brave://flags`

**Comment by pitsi** [comment]
It worked for me!

**Comment by ask1234560** [comment]
Thanks @antonok-edm it works for me too.

**Comment by antonok-edm** [comment]
@kjozwiak I added `QA/No` here since there's already a `QA/Yes` label on https://github.com/brave/brave-browser/issues/16283, which is tied to the same PR - feel free to update that if necessary

**Comment by pitsi** [comment]
When 1.27.x reaches stable and this thing is resolved, can someone please post here and let us know to change the flag back?

**Comment by antonok-edm** [comment]
@pitsi This change has been [uplifted](https://github.com/brave/brave-core/pull/9073) to the next 1.25.x hotfix release, which is actually scheduled for later today as far as I'm aware.

**Comment by pitsi** [comment]
Thanks! I got the upgrade a few minutes ago, switched `#brave-adblock-csp-rules` back to default and now works as it should.

**Comment by pitsi** [comment]
Sorry for bringing up a 3 month old issue, but I have to ask before opening a new one. A page I visit shows up blank since the 2 weeks ago and this is what the console says
`Access to internal resource at 'https://www.e-shop.gr/site.webmanifest' from origin 'https://e-shop.gr' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`
If it is not the same, I will open a new issue.

**Comment by ask1234560** [comment]
@pitsi the CORS error is not related to brave. CORS or cross origin request sharing error happens when CORS header is not present usually this is done in the backend.
